hollow/src/main/java/com/netflix/hollow/core/memory/HollowUnsafeHandle.java

```diff
@@ -38,10 +44,102 @@
             log.log(Level.SEVERE, "Unsafe access failed", e);
         }
         unsafe = u;
+        singleton = new HollowUnsafeHandle();
+    }
+
+    public static HollowUnsafeHandle getUnsafe() {
+        return singleton;
+    }
+
+    public void putOrderedLong(Object o, long l, long l1) {
+        if (UNALIGNED_ALLOWED) {
+            unsafe.putOrderedLong(o, l, l1);
+        } else {
+            // Assume little endian
+            unsafe.putByteVolatile(o, l, (byte) (l1));
+            unsafe.putByteVolatile(o, l + 1, (byte) (l1 >>> 8));
+            unsafe.putByteVolatile(o, l + 2, (byte) (l1 >>> 16));
+            unsafe.putByteVolatile(o, l + 3, (byte) (l1 >>> 24));
+            unsafe.putByteVolatile(o, l + 4, (byte) (l1 >>> 32));
+            unsafe.putByteVolatile(o, l + 5, (byte) (l1 >>> 40));
+            unsafe.putByteVolatile(o, l + 6, (byte) (l1 >>> 48));
+            unsafe.putByteVolatile(o, l + 7, (byte) (l1 >>> 56));
+        }
+    }
+
+    public long getLong(Object o, long l) {
+        if (UNALIGNED_ALLOWED) {
+            return unsafe.getLong(o, l);
+        } else {
+            // Assume little endian
+            return ((long) unsafe.getByte(o, l) & 0xFF)
```

Comment:
If we end up needing to do this, it will likely be faster to batch these up into- read 2 words when a read spans a word boundary then stitch up the long. 
If I am reading this right the `getByte`s translate into reading word or half word anyway https://developer.arm.com/documentation/ddi0360/f/unaligned-and-mixed-endian-data-access-support/unaligned-access-support/armv6-extensions

Reply:
Yeah, one of the points I mentioned was detecting when accesses are 1, 2, 4, or 8 bytes aligned and performing the most efficient one (any performance penalty from the branches should be outweighed by the gains from larger reads/writes)

At the time, I wasn't sure how to determine this, since all accesses are done with "object plus offset" methods - but it seems that heap objects are always 8-byte aligned, so we should be able to figure this out based on the offset alone. I'll update the PR with this optimization.

Reply:
Heads up the ByteBuffers in `EncodedByteBuffer` and `EncodedLongBuffer` are off heap and won't be word-aligned, I think